backend/app/api/routers/threads.py

```diff
@@ -0,0 +1,82 @@
+from datetime import datetime
+from http import HTTPStatus
+from typing import Optional
+
+from fastapi import APIRouter, WebSocket, WebSocketDisconnect
+
+from app.api.exceptions import EntityNotFound
+from app.api.managers import WsConnectionManager
+from app.api.repositories import SupportThreadRepository
+from app.api.schemas import (
+    Filter,
+    MessageCreate,
+    SupportThread,
+    SupportThreadCreate,
+    SupportThreadPatch,
+)
+
+router = APIRouter()
+ws_manager = WsConnectionManager()
+
+
+@router.post('/threads', response_model=SupportThread, status_code=HTTPStatus.CREATED)
+async def create_thread(thread: SupportThreadCreate) -> SupportThread:
+    thread_new = await SupportThreadRepository.create(thread)
+
+    return thread_new
+
+
+@router.get('/threads', response_model=list[SupportThread])
+async def find_all_threads(flt: Optional[Filter] = None) -> list[SupportThread]:
+    threads = await SupportThreadRepository.find_all(flt)
+
+    return threads
+
+
+@router.get('/threads/{thread_id}', response_model=SupportThread)
+async def find_thread(thread_id: int) -> SupportThread:
+    thread = await SupportThreadRepository.find_by_id(thread_id)
+
+    if thread is None:
+        raise EntityNotFound('thread')
+
+    return thread
+
+
+@router.patch('/threads/{thread_id}', response_model=SupportThread)
+async def patch_thread(thread_id: int, thread_ptc: SupportThreadPatch) -> SupportThread:
+    thread = await SupportThreadRepository.find_by_id(thread_id)
+
+    if thread is None:
+        raise EntityNotFound('thread')
+
+    thread = await SupportThreadRepository.patch(thread_ptc, thread_id)
+
+    return thread
+
+
+@router.delete('/threads/{thread_id}')
+async def delete_thread(thread_id: int) -> None:
+    thread = await SupportThreadRepository.find_by_id(thread_id)
+
+    if thread is None:
+        raise EntityNotFound('thread')
+
+    await SupportThreadRepository.delete(thread_id)
```

Comment:
Race condition of the same kind. What happens if I attempt to delete the same thread two times in parallel? Assume that they both find that the thread exists and then they both request to delete it (sequentially). Something bad will happen to one of them